SimPEG/electromagnetics/static/utils/static_utils.py

```diff
@@ -181,40 +198,62 @@ def source_receiver_midpoints(survey, **kwargs):
         )
 
     # Pre-allocate
-    midxy = []
-    midz = []
+    pseudo_locations = []
+
+    midpoints = []
+    ds = []
 
     for ii, source in enumerate(survey.source_list):
-        tx_locs = source.location
-        if isinstance(tx_locs, list):
-            Cmid = (tx_locs[0][:-1] + tx_locs[1][:-1]) / 2
-            zsrc = (tx_locs[0][-1] + tx_locs[1][-1]) / 2
-            tx_sep = np.linalg.norm((tx_locs[0][:-1] - tx_locs[1][:-1]))
+        src_loc = source.location
+        if isinstance(src_loc, list):
+            src_midpoint = (src_loc[0] + src_loc[1])/2
         else:
-            Cmid = tx_locs[:-1]
-            zsrc = tx_locs[-1]
+            src_midpoint = src_loc
+        src_midpoint = src_midpoint.reshape((1, len(src_midpoint)))
 
-        Pmids = []
         for receiver in source.receiver_list:
             rx_locs = receiver.locations
             if isinstance(rx_locs, list):
-                Pmid = (rx_locs[0][:, :-1] + rx_locs[1][:, :-1]) / 2
+                rx_midpoints = (rx_locs[0] + rx_locs[1])/2
             else:
-                Pmid = rx_locs[:, :-1]
-            Pmids.append(Pmid)
-        Pmid = np.vstack(Pmids)
-
-        midxy.append((Cmid + Pmid) / 2)
-        diffs = np.linalg.norm((Cmid - Pmid), axis=1)
-        if np.allclose(diffs, 0.0):  # likely a wenner type survey.
-            midz = zsrc - tx_sep / 2 * np.ones_like(diffs)
-        else:
-            midz.append(zsrc - diffs / 2)
+                rx_midpoints = rx_locs
+            n_loc = rx_midpoints.shape[0]
+
+            # Midpoint locations
+            midpoints.append(
+                (np.tile(src_midpoint, (n_loc, 1)) + rx_midpoints)/2
+            )
+
+            # Vector path from source midpoint to receiver midpoints
+            ds.append(
+                (rx_midpoints - np.tile(src_midpoint, (n_loc, 1)))
+            )
+    
+    midpoints = np.vstack(midpoints)
+    ds = np.vstack(ds)
+    pseudo_depth = np.zeros_like(midpoints)
+
+    # wenner-like electrode groups (are source and rx midpoints in same place)
+    is_wenner = np.sqrt(np.sum(ds[:,:-1]**2, axis=1)) < 0.1
```

Comment:
In a synthetic modeling scenario, the midpoints for the source electrodes and for the receiver electrodes would be identical. But in practice, it may not be the case. I wanted a threshold that was smaller than any reasonable electrodes spacing for a DCIP survey. But I wanted to make the threshold big enough just in case the locations of the Wenner electrodes wasn't perfect.

Reply:
let's say someone is wanting to plot a pseudo-section over some example (say in a small fish tank) where the electrodes them selves are only centimeters apart. This will incorrectly identify those points as Wenner.

Reply:
added a keyword argument to see tolerance for identifying wenner data